api/clients/disperser_client.go

```diff
@@ -37,47 +38,73 @@ func NewConfig(hostname, port string, timeout time.Duration, useSecureGrpcFlag b
 }
 
 type DisperserClient interface {
+	Close() error
 	DisperseBlob(ctx context.Context, data []byte, customQuorums []uint8) (*disperser.BlobStatus, []byte, error)
 	DisperseBlobAuthenticated(ctx context.Context, data []byte, customQuorums []uint8) (*disperser.BlobStatus, []byte, error)
 	GetBlobStatus(ctx context.Context, key []byte) (*disperser_rpc.BlobStatusReply, error)
 	RetrieveBlob(ctx context.Context, batchHeaderHash []byte, blobIndex uint32) ([]byte, error)
 }
 
+// See the NewDisperserClient constructor's documentation for details and usage examples.
 type disperserClient struct {
 	config *Config
 	signer core.BlobRequestSigner
+	// conn and client are not initialized in the constructor, but are initialized lazily
+	// whenever a method is called, using initOnce to make sure initialization happens only once
+	// and is thread-safe
+	initOnce sync.Once
+	conn     *grpc.ClientConn
+	client   disperser_rpc.DisperserClient
 }
 
 var _ DisperserClient = &disperserClient{}
 
-func NewDisperserClient(config *Config, signer core.BlobRequestSigner) DisperserClient {
+// DisperserClient maintains a single underlying grpc connection to the disperser server,
+// through which it sends requests to disperse blobs, get blob status, and retrieve blobs.
```

Comment:
probably should take care of the max concurrent requests, see https://learn.microsoft.com/en-us/aspnet/core/grpc/performance?view=aspnetcore-8.0#connection-concurrency. 

If it reaches the max, the new call get blocked. It can happen if OP uses concurrent blobs spinning up 100 conn

Reply:
Ah good catch. We can bump that limit or make it a configuration actually.
But I really don't think it's that big of a worry. OP would have to literally make 100 requests at the same time, which I think is unlikely? How long does a typical DisperseBlobAuthenticated take? < 5secs hopefully?
Also what happens if that limit is reached? it returns a 500 or something? Then batcher would just resubmit the blob a little bit later once the other connections are done processing.

Reply:
DisperseBlobAuthenticated takes about 300ms for 1MB blob

Reply:
Update comment: [ef1c7a5](https://github.com/Layr-Labs/eigenda/pull/826/commits/ef1c7a591109da15e9759020783b0cbb3821242d)